### PR TITLE
[SFEQS-1314]: EventHub integration for invoicing

### PR DESCRIPTION
**Pull request by manuelcoppotelli**
### Type of changes

- [x] Add new resources
- [ ] Update configuration to existing resources
- [ ] Remove existing resources

### Env to apply

- [ ] DEV
- [ ] UAT
- [x] PROD

### Does this introduce a change to production resources with possible user impact?

- [ ] Yes, users may be impacted applying this change
- [x] No

### Does this introduce an unwanted change on infrastructure? Check terraform plan execution result

- [ ] Yes
- [x] No

### Other information

Depends on #395

---

### How to apply

After PR is approved
1. run deploy pipeline from Azure DevOps [io-platform-iac-projects](https://dev.azure.com/pagopaspa/io-platform-iac-projects/_build?view=folders)
2. select PR branch
3. wait for approval

### CODE REVIEW

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
add zone redundant property for event hub
https://registry.terraform.io/providers/hashicorp/azurerm/2.99.0/docs/resources/eventhub_namespace#zone_redundant

**Review by pasqualedevita — CHANGES_REQUESTED** [review]
consider to use private endpoints for event hub

File: src/domains/sign/env/prod/terraform.tfvars [hunk 1]
```diff
@@ -73,3 +73,21 @@ io_sign_user_func = {
...
+  capacity = 1
+  hubs = {
+    billing = {
+      partition_count   = 2
+      message_retention = 1
+    }
+    institutions = {
+      partition_count   = 2
+      message_retention = 1
+    }
```
Comment (grausof) [inline src/domains/sign/env/prod/terraform.tfvars:87]: The documentation has been updated, we don't need this anymore.

File: src/domains/sign/env/prod/terraform.tfvars [hunk 2]
```diff
@@ -73,3 +73,21 @@ io_sign_user_func = {
   autoscale_minimum = 1
   autoscale_maximum = 5
 }
+
+integration_hub = {
+  capacity = 1
+  hubs = {
+    billing = {
+      partition_count   = 2
```
Comment (grausof) [inline src/domains/sign/env/prod/terraform.tfvars:81]: We need a partition count of 3 or multiple.

File: src/domains/sign/env/prod/terraform.tfvars [hunk 3]
```diff
@@ -73,3 +73,21 @@ io_sign_user_func = {
...
+      message_retention = 1
+    }
+    institutions = {
+      partition_count   = 2
+      message_retention = 1
+    }
+    analytics = {
+      partition_count   = 2
```
Comment (grausof) [inline src/domains/sign/env/prod/terraform.tfvars:89]: ditto

File: src/domains/sign/invoicing.tf [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+  message_retention   = var.integration_hub.hubs.billing.message_retention
+}
+
+resource "azurerm_eventhub" "iosign_integration_institutions" {
+  name                = "institutions"
+  namespace_name      = azurerm_eventhub_namespace.iosign_integration.name
+  resource_group_name = azurerm_resource_group.integration_rg.name
+  partition_count     = var.integration_hub.hubs.institutions.partition_count
+  message_retention   = var.integration_hub.hubs.institutions.message_retention
+}
```
Comment (grausof) [inline src/domains/sign/invoicing.tf:25]: ditto

File: src/domains/sign/io_sign_issuer_func.tf [hunk 5]
```diff
@@ -22,6 +22,7 @@ locals {
       IoServicesSubscriptionKey                       = module.key_vault_secrets.values["IOApiSubscriptionKey"].value
       PdvTokenizerApiBasePath                         = "https://api.uat.tokenizer.pdv.pagopa.it"
       PdvTokenizerApiKey                              = module.key_vault_secrets.values["TokenizerApiSubscriptionKey"].value
+      EventHubsConnectionString                       = module.key_vault_secrets.values["EventHubsConnectionString"].value
```
Comment (grausof) [inline src/domains/sign/io_sign_issuer_func.tf:25]: Renamed to `BillingEventHubConnectionString` and added `AnalyticsEventHubConnectionString` https://github.com/pagopa/io-sign/pull/73

Comment (manuelcoppotelli) [reply src/domains/sign/io_sign_issuer_func.tf]: Done 👍

File: src/domains/sign/env/prod/terraform.tfvars [hunk 6]
```diff
@@ -73,3 +80,32 @@ io_sign_user_func = {
...
+    }
+    institutions = {
+      partition_count   = 3
+      message_retention = 1
+    }
+    analytics = {
+      partition_count   = 3
+      message_retention = 1
+    }
+  }
```
Comment (grausof) [inline src/domains/sign/env/prod/terraform.tfvars:110]:
```suggestion
    billing = {
      partition_count   = 3
      message_retention = 7
    }
    institutions = {
      partition_count   = 3
      message_retention = 7
    }
    analytics = {
      partition_count   = 3
      message_retention = 7
    }
  }
```

**Review by grausof — CHANGES_REQUESTED** [review]

**Comment by manuelcoppotelli** [comment]
@pagopa/io-backend-admin Could you please review it?

File: src/domains/sign/env/prod/terraform.tfvars [hunk 7]
```diff
@@ -73,3 +80,66 @@ io_sign_user_func = {
   autoscale_minimum = 1
   autoscale_maximum = 5
 }
+
+integration_hub = {
+  auto_inflate_enabled     = true
+  sku_name                 = "Standard"
+  capacity                 = 1
```
Comment (grausof) [inline src/domains/sign/env/prod/terraform.tfvars:87]: it's enough?

**Review by grausof — APPROVED** [review]
LGTM!

**Comment by pasqualedevita** [comment]
same change done in https://github.com/pagopa/io-infra/pull/417

**Pull request closed without merging**